source/secure_tunneling_operations.c

```diff
@@ -138,30 +189,83 @@ void aws_secure_tunnel_message_view_log(
         (void *)message_view,
         aws_secure_tunnel_message_type_to_c_string(message_view->type));
 
-    if (message_view->service_id != NULL) {
-        AWS_LOGUF(
-            log_handle,
-            level,
-            AWS_LS_IOTDEVICE_SECURE_TUNNELING,
-            "id=%p: aws_secure_tunnel_message_view service_id set to '" PRInSTR "'",
-            (void *)message_view,
-            AWS_BYTE_CURSOR_PRI(*message_view->service_id));
-    } else {
-        AWS_LOGUF(
-            log_handle,
-            level,
-            AWS_LS_IOTDEVICE_SECURE_TUNNELING,
-            "id=%p: aws_secure_tunnel_message_view service_id not set",
-            (void *)message_view);
-    }
+    switch (message_view->type) {
+        case AWS_SECURE_TUNNEL_MT_DATA:
+        case AWS_SECURE_TUNNEL_MT_STREAM_START:
+        case AWS_SECURE_TUNNEL_MT_STREAM_RESET:
+        case AWS_SECURE_TUNNEL_MT_CONNECTION_START:
+        case AWS_SECURE_TUNNEL_MT_CONNECTION_RESET:
+            if (message_view->service_id != NULL) {
+                AWS_LOGUF(
+                    log_handle,
+                    level,
+                    AWS_LS_IOTDEVICE_SECURE_TUNNELING,
+                    "id=%p: aws_secure_tunnel_message_view service_id set to '" PRInSTR "'",
+                    (void *)message_view,
+                    AWS_BYTE_CURSOR_PRI(*message_view->service_id));
```

Comment:
Only to the extent that the customer sets them up as something sensitive. Knowing what service id a secure tunnel has is useless without the access token and the client token which are unique for the destination and source. If we don't log service ids, it would significantly reduce the usefulness of the related logs. We may also need to not log Connection Ids in that case as the user could also set those to be sensitive. The expectation is these are set to pretty basic and mundane names like "ssh" or "http".

Reply:
Thought a bit more on this and we should be safe. The first thing the secure tunnel service does when a source or destination device connects to it is emit the available service ids to whatever has connected using the access/client tokens.